samcli/local/apigw/authorizers/lambda_authorizer.py

```diff
@@ -0,0 +1,254 @@
+            The regular expression that can be used to validate headers
+        use_simple_responses: bool = False
+            Boolean representing whether to return a simple response or not
+        """
+        self.authorizer_name = authorizer_name
+        self.lambda_name = lambda_name
+        self.type = type
+        self.validation_string = validation_string
+        self.payload_version = payload_version
+        self.use_simple_response = use_simple_response
+
+        self._parse_identity_sources(identity_sources)
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, LambdaAuthorizer)
+            and self.lambda_name == other.lambda_name
+            and sorted(self._identity_sources_raw) == sorted(other._identity_sources_raw)
+            and self.validation_string == other.validation_string
+            and self.use_simple_response == other.use_simple_response
+            and self.payload_version == other.payload_version
+            and self.authorizer_name == other.authorizer_name
+            and self.type == other.type
+        )
+
+    @property
+    def identity_sources(self) -> List[IdentitySource]:
+        """
+        The list of identity source validation objects
+
+        Returns
+        -------
+        List[IdentitySource]
+            A list of concrete identity source validation objects
+        """
+        return self._identity_sources
+
+    @identity_sources.setter
+    def identity_sources(self, identity_sources: List[str]) -> None:
+        """
+        Parses and sets the identity source validation objects
+
+        Parameters
+        ----------
+        identity_sources: List[str]
+            A list of strings of identity sources
+        """
+        self._parse_identity_sources(identity_sources)
+
+    def _parse_identity_sources(self, identity_sources: List[str]) -> None:
+        """
+        Helper function to create identity source validation objects
+
+        Parameters
+        ----------
+        identity_sources: List[str]
+            A list of identity sources to parse
+        """
+        identity_source_type = {
+            "method.request.header.": HeaderIdentitySource,
+            "$request.header.": HeaderIdentitySource,
+            "method.request.querystring.": QueryIdentitySource,
+            "$request.querystring.": QueryIdentitySource,
+            "context.": ContextIdentitySource,
+            "$context.": ContextIdentitySource,
+            "stageVariables.": StageVariableIdentitySource,
+            "$stageVariables.": StageVariableIdentitySource,
+        }
+
+        self._identity_sources_raw = identity_sources
+        self._identity_sources = []
+
+        for identity_source in self._identity_sources_raw:
+            for prefix, identity_source_object in identity_source_type.items():
+                if identity_source.startswith(prefix):
+                    # get the stuff after the prefix
```

Comment:
what should we do if the provided identity source is not one of the known Identity sources?

Reply:
As of right now, it doesn't do anything if there is an identity source that doesn't match the above table. In the instances where we create Lambda Authorizers, identity sources are already validated before being passed in, however this method could be a bit more defensive.

I'll add some checks to validate that the incoming identity sources are valid.